entitlements/entitlements.go

```diff
@@ -56,3 +58,67 @@ var AllEntitlements = []EntitlementKind{
 	ExternalAuditStorage, FeatureHiding, HSM, Identity, JoinActiveSessions, K8s, MobileDeviceManagement, OIDC, OktaSCIM,
 	OktaUserSync, Policy, SAML, SessionLocks, UpsellAlert, UsageReporting,
 }
+
+// supportEntitlementsCompatibility ensures entitlements are backwards compatible
+// If Entitlements are present, there are no changes
+// If Entitlements are not present, sets the entitlements fields to legacy field values
+// TODO(michellescripts) remove in v18
+func SupportEntitlementsCompatibility(features *proto.Features) {
```

Comment:
Let's rename this - now that it's in the entitlements package we don't need to say "entitlements" twice.

How about calling it `entitlements.BackfillFeatures`?